anomalib/models/cflow/config.yaml

```diff
@@ -55,10 +55,12 @@ trainer:
   benchmark: false
   check_val_every_n_epoch: 1
   default_root_dir: null
+  detect_anomaly: false
   deterministic: false
   enable_checkpointing: true
+  enable_progress_bar: true
   fast_dev_run: false
-  gpus: 1
+  gpus: null # Set automatically
```

Comment:
Does it mean that pytorch lightning will distribute job across all available GPUs? In which case, we might have to look at the performance.

Reply:
Not sure what's the best approach here. Setting it to 1 or automatically assigning it?
@ashwinvaidya17, @djdameln?

Reply:
I would prefer using only a single gpu by default. I feel users should use distribute only if they know what they are doing. If we distribute the training then we will have to look at the learning rate and experiments might not be reproducible across different number of GPUs.

Reply:
What if no GPUs are available. Will it break or just switch to CPU training?

Reply:
> What if no GPUs are available. Will it break or just switch to CPU training?

It switches to CPU.

> I would prefer using only a single gpu by default. I feel users should use distribute only if they know what they are doing. If we distribute the training then we will have to look at the learning rate and experiments might not be reproducible across different number of GPUs.

As far as I understand, this doesn't set it to distributed training because `strategy` is still `null`. It automatically uses a single GPU.

Reply:
In that case I am fine with how it is now.